### ISSUE TITLE
.Net 2.0 is no longer a reasonable requirement

### ISSUE THREAD

**Issue by TaoK** [TaoK/PoorMansTSqlFormatter#199]
Users have had trouble with the Notepad++ plugin, and more recently apparently also the SSMS Add-In, because it requires .Net 2.0.

This version of .Net is no longer installed on windows machines by default, and it's not at all obvious how to get it, when the installer tells you you need it and don't have it.

For reference, and as noted in Issue #157, in Windows 10 the .Net Framework 3.5 SP1 (including .Net 2.0, and so enabling the NPP plugin) is an optional feature of windows, that can be enabled in the "Turn Windows features on or off" screen/functionality. It's called ".NET Framework 3.5 (includes .NET 2.0 and 3.0)", and it's right at the top of the list of optional features.

**Comment by joliverSCSO** [comment]
I also can not longer use this formatter due to the .Net 2.0 requirement.

**Comment by dekenless** [comment]
Sadly, this is preventing us from using PoorMansTSqlFormatter any longer after moving to Windows 10. I think priority should be given to eliminating the .NET Framework 2.0 dependency.

**Comment by AeroXuk** [comment]
Switching to the DotNetCore style project files would allow you to target multiple .NET Framework versions simultaneously.  Allowing you to keep .NET Framework v2 capability and also target a slightly higher default version (maybe 4.0).  And also target the latest .NET Standard 2.0 within class libraries.

**Comment by EmoPlus** [comment]
Signed up to GitHub just to up this as this is causing me a problem also and I enjoy this formatter.

**Comment by mhall1-new** [comment]
+1 to this!

A recent update broke the ability to even download .NET 3.5 through the GUI or the Microsoft site. The only way to do so now is to find the package source on your Windows 10 installation media (or online if you're lucky) and run the following:

dism /online /add-package /packagepath:C:\blahblahblah\microsoft-windows-netfx3-ondemand-package.cab

That's a big barrier to entry.

**Comment by talent-hctw** [comment]
Hi,
It's been 3 years passed.
Do we still need .Net 2.0?
I just tried to install it on SSMS 18.8, and same error pop-up.
Is there any plan to upgrade it?

**Comment by OwnerOfThisIsle** [comment]
+1 
Installing .Net 2.0 seems no easy task on a current windows. So pleaaaase, remove this requirement / update it to a newer .NET

**Comment by walidhaidarii** [comment]
Please upgrade the .netframework to 4+

**Comment by cullen-sloboda** [comment]
i agree. now my plugin to npp+ broke too. can't use this product at all
